core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -75,7 +75,20 @@
   public static final String MAX_FILE_SIZE_BYTES = "max-file-size-bytes";
   public static final double MAX_FILE_SIZE_DEFAULT_RATIO = 1.80d;
 
+  /**
+   * The minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting.
+   * If a data file has more than this number of deletes, it will be rewritten regardless of its file size determined
+   * by {@link #MIN_FILE_SIZE_BYTES} and {@link #MAX_FILE_SIZE_BYTES}.
+   * If a file group contains a file that satisfies this condition, the file group will be rewritten regardless of
+   * the number of files in the file group determined by {@link #MIN_INPUT_FILES}
+   * <p>
+   * Defaults to Integer.MAX_VALUE, which means this feature is not enabled by default.
+   */
+  public static final String MIN_DELETES_PER_FILE = "min-deletes-per-file";
```

Comment:
What value is recommended to the user? Or how does the user compute the proper value? I'm thinking maybe adding an option to count filegroup valid if any file of it contains deletes. Because you don't know how many records match the equality delete, for example, delete a set of records in an area/province.

Nit: This is disabled by default but it is always comparing.

Reply:
I think it's fine to just do this based on the number of files since the read penalty is directly related to the number of files and less so to the amount of actual rows deleted. 

No strong feeling on the default since we already have the amount of delete files in the task information so I don't think the check is very expensive

Reply:
Also, do we need to have specific settings for equality delete and position delete separately? Since the read penalties are different.

Reply:
I think that is equivalent to setting this value to 0?

Reply:
I think the exact value to set depends on the user's tolerance of read performance, because more deletes means worse read performance and potentially getting out of memory, so users can tune this value based on their system requirements.

Reply:
> I think that is equivalent to setting this value to 0?

Hm, that's it.